lib/git.js

```diff
@@ -51,4 +51,20 @@ module.exports.remote = async function remote (opts = {}) {
   } catch (e) {
     // ignore error
   }
+}
+module.exports.remote = gitRemote;
+
+module.exports.repository = async function (cwd, pkg) {
+  let repo;
+  if (!pkg || !pkg.repository) {
+    const remote = await gitRemote({ cwd });
+    if (remote) {
+      repo = remote;
+    }
+  } else if (pkg && typeof pkg.repository === 'string') {
+    repo = pkg.repository;
+  } else if (pkg && typeof pkg.repository !== 'undefined' && pkg.repository.url) {
+    repo = pkg.repository.url;
+  }
+  return repo;
 };
```

Comment:
I am wondering now if we can delete all this since `@npmcli/package-json` does more normalization?